src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFileFormat.cs

```diff
@@ -630,15 +630,15 @@ private static JProperty WritePackageDependencyGroup(PackageDependencyGroup item
                 WriteObject(item.Packages, WritePackageDependency));
         }
 
-        private static PackageDependency ReadPackageDependency(string property, JToken json)
+        internal static PackageDependency ReadPackageDependency(string property, JToken json)
```

Comment:
You can probably extract all the common methods to a different class than have this weird dependency from one lock file to another one.

Reply:
It will be more weird then because then we're talking about common class of 2 different lock file formats. At least here `LockFileFormat` has all the details about its own format and other file is using that same understanding since that other file is build from this `LockFileFormat`